tt_metal/impl/dispatch/arch.cpp

```diff
@@ -0,0 +1,738 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "kernel_config/fd_kernel.hpp"
+#include "impl/device/device_pool.hpp"
+#include "tt_metal/detail/tt_metal.hpp"
+#include "kernel_config/fd_kernel.hpp"
+#include "kernel_config/prefetch_kernel.hpp"
+#include "kernel_config/dispatch_kernel.hpp"
+#include "kernel_config/dispatch_s_kernel.hpp"
+#include "kernel_config/mux_kernel.hpp"
+#include "kernel_config/demux_kernel.hpp"
+#include "kernel_config/eth_router_kernel.hpp"
+#include "kernel_config/eth_tunneler_kernel.hpp"
+
+#define DISPATCH_MAX_UPSTREAM 4
+#define DISPATCH_MAX_DOWNSTREAM 4
+
+typedef struct {
+    int id;
+    int device_id;
+    uint8_t cq_id;
+    DispatchWorkerType kernel_type;
+    int upstream_ids[DISPATCH_MAX_UPSTREAM];
+    int downstream_ids[DISPATCH_MAX_DOWNSTREAM];
+    NOC my_noc;
+    NOC upstream_noc;
+    NOC downstream_noc;
+} dispatch_kernel_node_t;
+
+// For readablity, unset = x = -1
+#define x -1
+
+void increment_node_ids(dispatch_kernel_node_t &node, uint32_t inc) {
+    node.id += inc;
+    for (int &id : node.upstream_ids)
+        if (id != x)
+            id += inc;
+    for (int &id : node.downstream_ids)
+        if (id != x)
+            id += inc;
+}
+
+static const std::vector<dispatch_kernel_node_t> single_card_arch_1cq = {
```

Comment:
instead of *_card_* can we call name these *_chip_* or *_device_* because we could have an a single n300 that has 2 chips for e.g.

Reply:
Yeah it should be chip and not card, will fix this 👍